crates/fluvio-stream-model/src/core.rs

```diff
@@ -143,17 +147,23 @@ mod core_model {
 
     use std::fmt::{Debug, Display};
     use std::hash::Hash;
+    use std::str::FromStr;
+
+    use serde::Serialize;
+    use serde::de::DeserializeOwned;
 
     /// metadata driver
     pub trait MetadataStoreDriver {
         type Metadata;
     }
 
-    pub trait Spec: Default + Debug + Clone + PartialEq + Send + Sync + 'static {
+    pub trait Spec:
+        Default + Debug + Clone + PartialEq + Serialize + DeserializeOwned + Send + Sync + 'static
+    {
         const LABEL: &'static str;
-        type Status: Status;
+        type Status: Status + Serialize + DeserializeOwned;
```

Comment:
Unfortunately, we don't have many choices out here. We either:
1. make `Spec` Serialize + Deserialize
2. create an intermediate type for every spec to convert from it just for the sake of serialization/deserialization. And another `ExtendedSpec` that will hold a link between the Spec and the new type. A lot of manual work and a lot of code to pack back and forth.

Effectively, all specs already implement Serialize + DeserializeOwned, so this is just a fixation of this fact.

Reply:
In this kind of case, you can use auto trait like here https://docs.rs/futures-util/0.3.28/src/futures_util/stream/stream/mod.rs.html#249.
```
impl<T: ?Sized> StreamExt for T where T: Stream {}
```
So there is no need for manual implementation.

Reply:
Actually, I read thru code and realize it's fine as is since it's under `serde` config already.  Adding more auto traits just add more unnecessary abstraction